README.md

```diff
@@ -141,16 +141,18 @@ in a collection of configuration options.  The defaults the SDK picks are sensib
-| ------------------------ | ------- | ------------------------------------------------------------ | ------- |
-| `cacheIdentity`          | boolean | Caches the identity of users in the browser's local storage to capture cross-session behaviours | `true`  |
-| `trackReferrer`          | boolean | Whether or not to emit an initial `REFERRER` event containing the referrer attribute | `true`  |
-| `trackPages`             | boolean | Tracks whenever there is a URL change during the session and logs it automatically. | `true`  |
-| `trackUTM`               | boolean | Automatically reports the UTM tags (`utm_campaign, utm_medium, utm_source`) of the first page visit | `true`  |
-| `trackWalletConnections` | boolean | Automatically track wallet connections. Currently only supporting Metamask. | `true`  |
-| `trackChainChanges`      | boolean | Automatically track chain ID changes. Currently only supporting Metamask | `true`  |
-| `trackTransactions`      | boolean | Automatically track transaction requests before they are sent to Metamask. Currently only supporting Metamask | `true`  |
-| `trackSigning`           | boolean | Automatically track signing requests before they are signed in Metamask. Currently only supporting Metamask. | `true`  |
+| Config key               | Type            | Description                                                  | Default           |
+| ------------------------ | --------------- | ------------------------------------------------------------ | ----------------- |
+| `cacheIdentity`          | boolean         | Caches the identity of users in the browser's local storage to capture cross-session behaviours | `true`            |
+| `initialProvider`        | EIP1193Provider | The provider to use for the web3 tracking events             | `window.ethereum` |
+| `trackReferrer`          | boolean         | Whether or not to emit an initial `REFERRER` event containing the referrer attribute | `true`            |
+| `trackPages`             | boolean         | Tracks whenever there is a URL change during the session and logs it automatically. | `true`            |
+| `trackUTM`               | boolean         | Automatically reports the UTM tags (`utm_campaign, utm_medium, utm_source`) of the first page visit | `true`            |
+| `trackWalletConnections` | boolean         | Automatically track wallet connections on the provider passed to `initialProvider` or `setProvider`. | `true`            |
+| `trackChainChanges`      | boolean         | Automatically track chain ID changes on the provider passed to `initialProvider` or `setProvider`. | `true`            |
+| `trackTransactions`      | boolean         | Automatically track transaction requests  on the provider passed to `initialProvider` or `setProvider`. | `true`            |
+| `trackSigning`           | boolean         | Automatically track signing requests  on the provider passed to `initialProvider` or `setProvider`. | `true`            |
```

Comment:
that's not true for 100% because if you don't set `initialProvider and don't use `setProvider` `AnalyticsSdk` will still track because we get provider from window then

Reply:
I said that because I put `initialProvider`'s default value to 
<img width="837" alt="image" src="https://user-images.githubusercontent.com/5834876/208150208-7bf4b706-e541-4269-b4b1-9ccac49ed8c7.png">